stream_buffer.c

```diff
@@ -1171,25 +1184,20 @@ static size_t prvWriteBytesToBuffer( StreamBuffer_t * const pxStreamBuffer,
         mtCOVERAGE_TEST_MARKER();
     }
 
-    pxStreamBuffer->xHead = xNextHead;
-
-    return xCount;
+    return xNextHead;
 }
 /*-----------------------------------------------------------*/
 
 static size_t prvReadBytesFromBuffer( StreamBuffer_t * pxStreamBuffer,
                                       uint8_t * pucData,
-                                      size_t xMaxCount,
-                                      size_t xBytesAvailable )
+                                      size_t xCount,
+                                      size_t xTail )
 {
-    size_t xCount, xFirstLength, xNextTail;
-
-    /* Use the minimum of the wanted bytes and the available bytes. */
-    xCount = configMIN( xBytesAvailable, xMaxCount );
+    size_t xFirstLength, xNextTail;
 
     if( xCount > ( size_t ) 0 )
     {
-        xNextTail = pxStreamBuffer->xTail;
+        xNextTail = xTail;

```

Comment:
I think this line (setting xNextTail) needs to be outside the if() statement, otherwise when xCount == 0 it is returned without being initialised.

Reply:
Oof, I didn't even notice that compiler warning...